Tests/BackendIntegrationTests/TrialOrIntroEligibilityCheckerIntegrationTests.swift

```diff
@@ -0,0 +1,104 @@
+                                                               systemInfo: systemInfo))
+        self.checker = .init(
+            receiptFetcher: .init(requestFetcher: .init(operationDispatcher: .default),
+                                  systemInfo: systemInfo),
+            introEligibilityCalculator: .init(productsManager: .init(systemInfo: systemInfo),
+                                              receiptParser: .init()),
+            backend: backend,
+            currentUserProvider: currentUserProvider,
+            operationDispatcher: .default,
+            productsManager: .init(systemInfo: systemInfo)
+        )
+    }
+
+    func checkEligibility(_ productIdentifiers: [String]) async throws -> [String: IntroEligibility] {
+        fatalError("This method must be overriden")
+    }
+
+    /// This test is similar to ``StoreKit1IntegrationTests/testEligibleForIntroBeforePurchaseAndIneligibleAfter``
+    /// except ran directly on `TrialOrIntroPriceEligibilityChecker`, which makes the test force
+    /// either the SK1 or SK2 implementation, so they can be tested independently.
+    final func runTestEligibleForIntroBeforePurchaseAndIneligibleAfter() async throws {
+        let productID = try await self.monthlyPackage.storeProduct.productIdentifier
+
+        var eligibility = try await self.checkEligibility([productID])
+        expect(eligibility[productID]?.status) == .eligible
+
+        try await self.purchaseMonthlyOffering()
+
+        let anonUserID = Purchases.shared.appUserID
+        let identifiedUserID = "\(#function)_\(anonUserID)_".replacingOccurrences(of: "RCAnonymous", with: "")
+
+        let (identifiedCustomerInfo, created) = try await Purchases.shared.logIn(identifiedUserID)
+        expect(created) == true
+        expect(identifiedCustomerInfo.entitlements[Self.entitlementIdentifier]?.isActive) == true
+
+        eligibility = try await self.checkEligibility([productID])
+        expect(eligibility[productID]?.status) == .ineligible
+    }
+
+}
+
+class TrialOrIntroEligibilityCheckerStoreKit1IntegrationTests: BaseTrialOrIntroEligibilityCheckerIntegrationTests {
+
+    override func checkEligibility(_ productIdentifiers: [String]) async throws -> [String: IntroEligibility] {
+        return await withCheckedContinuation { contination in
+            self.checker.sk1CheckEligibility(productIdentifiers) { eligibility in
+                contination.resume(returning: eligibility)
+            }
+        }
+    }
+
+    func testEligibleForIntroBeforePurchaseAndIneligibleAfter() async throws {
+        try await self.runTestEligibleForIntroBeforePurchaseAndIneligibleAfter()
+    }
+}
+
+class TrialOrIntroEligibilityCheckerStoreKit2IntegrationTests: BaseTrialOrIntroEligibilityCheckerIntegrationTests {
+
+    override func checkEligibility(_ productIdentifiers: [String]) async throws -> [String: IntroEligibility] {
+        guard #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) else {
```

Comment:
Not using `AvailabilityChecks` so that I can get away with not marking this method with `@available`, that way the shared test can simply run on all classes.